### DISCUSSION TITLE
internal document links

### DISCUSSION THREAD

**Discussion by c-xlenz** [hrynko/vue-pdf-embed#50] [Q&A] [answered]
How can I jump to the page that is referenced in the pdf.

I get the callback... I can read the page and set it, but then it is ONLY this page and no way back.

It does not automatically jump to the page...

Any ideas?
(e.g. a gotoPage()...)

**Comment by hrynko** [comment, marked as answer]
This depends on whether you are displaying the document page by page or the entire document.

1. In the first case, you just need to specify the page from the `internal-link-clicked` event as the `page` prop.
2. In the second case, you can use a template ref or ID as in the example below:

```vue
<template>
  <vue-pdf-embed
    id="pdfId"
    ref="pdfRef"
    :source="pdfSource"
    @internal-link-clicked="handleLinkClick"
  />
</template>

<script>
import VuePdfEmbed from 'vue-pdf-embed'

export default {
  components: {
    VuePdfEmbed,
  },
  data() {
    return {
      pdfSource: '<PDF_SOURCE>',
    }
  },
  methods: {
    handleLinkClick(page) {
      this.$refs.pdfRef.$el.children[page - 1].scrollIntoView()
      // OR USING ID:
      // document.getElementById(`pdfId-${page}`).scrollIntoView()
    },
  },
}
</script>
```

**Reply by c-xlenz** [reply]
Fanstastic.
Thank you!
Do you also have a way to get the page number currently shown?

**Reply by hrynko** [reply]
I would subscribe to a `scroll` event and calculate the current page by the scroll position and page height.
